src/types/websockets/client.ts

```diff
@@ -23,6 +25,7 @@ export interface WSClientConfigurableOptions {
 
   wsUrl?: string;
 
+  agent?: Agent | undefined;
```

Comment:
Apologies for the delayed review. I wanted to double check how this was introduced in one of my other SDKs, so the pattern is consistent with all of them. First SDK to see the option to pass a custom agent to the WS connection, is the binance SDK. Could you update this to follow the same pattern?

Here's the configuration interface:
https://github.com/tiagosiebler/binance/blob/master/src/websocket-client.ts#L55-L58

Here's how the parameter is passed to the WebSocket constructor:
https://github.com/tiagosiebler/binance/blob/master/src/websocket-client.ts#L228-L230

You might notice some differences vs how you've passed an object as the second parameter (which seems incorrect). The second parameter should be the "protocols" array, while the 3rd parameter is everything else (including agent). This gives support for both the browser and node environments without conflicts

Browser spec:
https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket

Node spec:
https://github.com/websockets/ws/blob/master/doc/ws.md#new-websocketaddress-protocols-options

Should be a mostly copy-paste from the binance implementation, but let me know if you have questions!

Reply:
Thank you for the detailed feedback! I’ve reviewed the Binance SDK implementation and updated the code to follow the same pattern for passing the custom agent to the WebSocket constructor. The parameters have been adjusted to correctly use the "protocols" array as the second parameter, and the agent (along with other options) is now passed as the third parameter, aligning with both browser and Node.js specifications.

I’ve made the changes and pushed the update. Let me know if you have any further questions or suggestions.